dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/notification/DefaultProgramNotificationTemplateStore.java

```diff
@@ -39,11 +39,13 @@
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
 
 /**
  * Created by zubair@dhis2.org on 16.11.17.
  */
 @Repository( "org.hisp.dhis.program.notification.ProgramNotificationTemplateStore" )
+@Transactional
```

Comment:
What is the reason to move the `@Transactional` boundary "down" to the DAO class? Normally, we transactionally annotate services rather than a repository. I would suggest adding a short comment to explain the deviation from the norm.

Reply:
fixed by creating a service layer class